File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/facade/transport/Car.java
// Car.java
// All function implementations are dummy implementations, just to demonstrate Facade

package com.bigboxcode.designpattern.facade.transport;

import java.util.Random;

public class Car {

    public void startEngine() {
        System.out.println("Start Engine");
    }

    public void stopEngine() {
        System.out.println("Stop Engine");
    }

    public void goStraight() {
        System.out.println("Go Straight: ↑");
    }

    public void goLeft() {
        System.out.println("Go Left: ←");
    }

    public void goRight() {
        System.out.println("Go Right: →");
    }

    public double getDistanceTravelled() {
        Random r = new Random();
        return (r.nextInt((int)((10000-100)*10+1))+100*10) / 10.0;
    }
}

File: design-patterns/php/facade/travelplan/Car.php
<?php
// Car.php

namespace BigBoxCode\DesignPattern\Facade\TravelPlan;

class Car {
    public function startEngine(): void {
        echo "Start Engine\n";
    }

    public function stopEngine(): void {
        echo "Stop Engine\n";
    }

    public function goStraight(): void {
        echo "Go Straight: ↑\n";
    }

    public function goLeft(): void {
        echo "Go Left: ←\n";
    }

    public function goRight(): void {
        echo "Go Right: →\n";
    }

    public function getDistanceTravelled(): float {
        // This is some random calculation for demo purpose
        return ((rand(0, 180) * ((10000 - 100) * 10 + 1)) + 100 * 10) / 10.0;
    }
}